engine/access/state_stream/handler_test.go

```diff
@@ -110,6 +112,114 @@ func TestExecutionDataStream(t *testing.T) {
 	assert.Equal(t, 1, receivedCount)
 }
 
+// TestExecutionDataStreamEventEncoding tests the event encoding behavior of the execution data stream.
+func TestExecutionDataStreamEventEncoding(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Generate sample events for testing, each with a payload.
+	ccfEventGenerator := generator.EventGenerator(generator.WithEncoding(generator.EncodingCCF))
+	ccfEvents := make([]flow.Event, 0, 1)
+	ccfEvents = append(ccfEvents, ccfEventGenerator.New())
+
+	jsonEventsGenerator := generator.EventGenerator(generator.WithEncoding(generator.EncodingJSON))
+	jsonEvents := make([]flow.Event, 0, 1)
+	jsonEvents = append(jsonEvents, jsonEventsGenerator.New())
+
+	// Create a mock API and a mock stream for subscription.
+	api := ssmock.NewAPI(t)
+	stream := makeStreamMock[executiondata.SubscribeExecutionDataRequest, executiondata.SubscribeExecutionDataResponse](ctx)
+
+	// Helper function to perform a stream request and handle responses.
+	makeStreamRequest := func(request *executiondata.SubscribeExecutionDataRequest) {
+		sub := state_stream.NewSubscription(1)
+
+		api.On("SubscribeExecutionData", mock.Anything, flow.ZeroID, uint64(0), mock.Anything).Return(sub)
+
+		h := state_stream.NewHandler(api, flow.Localnet.Chain(), state_stream.EventFilterConfig{}, 1)
+
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			wg.Done()
+			err := h.SubscribeExecutionData(request, stream)
+			require.NoError(t, err)
+			t.Log("subscription closed")
+		}()
+		wg.Wait()
+
+		// Send a single response.
+		blockHeight := uint64(1)
+		executionData := unittest.BlockExecutionDataFixture(
+			unittest.WithChunkExecutionDatas(
+				unittest.ChunkExecutionDataFixture(t, 1024, unittest.WithChunkEvents(ccfEvents)),
+			),
+		)
+
+		err := sub.Send(ctx, &state_stream.ExecutionDataResponse{
+			Height:        blockHeight,
+			ExecutionData: executionData,
+		}, 100*time.Millisecond)
+		require.NoError(t, err)
+
+		// Notify end of data.
+		sub.Close()
+	}
+
+	// Test scenario for JSON event encoding.
+	t.Run("test JSON event encoding", func(t *testing.T) {
+		makeStreamRequest(&executiondata.SubscribeExecutionDataRequest{
+			EventEncodingVersion: entities.EventEncodingVersion_JSON_CDC_V0,
+		})
+		for {
+			resp, err := stream.RecvToClient()
+			if err == io.EOF {
+				break
```

Comment:
how do we hit this condition? If test just breakes here it's still a success even though no entries were actually asserted.